### PR TITLE
Media reorder logic

### PR DESCRIPTION
**Pull request by soupette**
#### Description of what you did:
This PR enable assets ordering using a dnd feature

### CODE REVIEW

File: packages/strapi-plugin-upload/admin/src/components/DragLayer/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        return null;
+    }
+  };
+
+  if (!isDragging) {
+    return null;
+  }
+
+  return (
+    <div style={layerStyles}>
```
Comment (HichamELBSI) [inline packages/strapi-plugin-upload/admin/src/components/DragLayer/index.js]: This should be a component.

File: packages/strapi-plugin-upload/admin/src/containers/App/index.js [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 const App = () => {
   return (
-    <Switch>
-      <Route path={`/plugins/${pluginId}`} component={HomePage} />
-    </Switch>
+    <DndProvider backend={HTML5Backend}>
+      <Switch>
+        <Route path={`/plugins/${pluginId}`} component={HomePage} />
+      </Switch>
+    </DndProvider>
```
Comment (HichamELBSI) [inline packages/strapi-plugin-upload/admin/src/containers/App/index.js]: Why do we need a second dndprovider here?

File: packages/strapi-plugin-upload/admin/src/components/Card/index.js [hunk 3]
```diff
@@ -28,15 +32,59 @@ const Card = ({
...
+    preview(getEmptyImage(), { captureDraggingState: true });
+  }, [preview]);
+
+  const opacity = isDragging ? 0.2 : 1;
+
   const handleClick = () => {
     onClick(id);
   };
 
+  drag(drop(ref));
```
Comment (HichamELBSI) [inline packages/strapi-plugin-upload/admin/src/components/Card/index.js]: All the logic of this file will be implemented in each card even if cards are not draggable (In the BrowseAsset tab and in the Plugin). I think we should create a DraggableCard that will be used only in the SelectedAssets list.

File: packages/strapi-plugin-upload/admin/src/containers/App/index.js (hunk 2 above)
Comment (soupette) [reply packages/strapi-plugin-upload/admin/src/containers/App/index.js]: Because we need one for the plugin for the content manager.

Comment (soupette) [reply packages/strapi-plugin-upload/admin/src/containers/App/index.js]: But now that I have created another component for the dnd as suggested it's not needed anymore.

**Pull request merged**